pages/gallery/page.js

```diff
@@ -0,0 +1,180 @@
+import { getMetadata } from '../../scripts/lib-franklin.js';
+
+let imageBlockCount = 0;
+let currentIndex = 0;
+
+function decorateMetadataContent(containerEL, key) {
+  const value = getMetadata(key);
+  if (value) {
+    containerEL.innerHTML = value;
+  }
+}
+
+function decorateAttribution(attributionEL) {
+  const authorMetadata = getMetadata('author');
+  const authorUrlMetadata = getMetadata('author-url');
+  if (authorMetadata) {
+    const names = authorMetadata.split(',');
+    const urls = (authorUrlMetadata) ? authorUrlMetadata.split(',') : [];
+
+    let i = 0;
+    let s = '';
+    names.forEach((n) => {
+      let span = null;
+      if (i < urls.length) {
+        span = `<span class="attribution-name"><a href="${urls[i]}">${n}</a></span>`;
+      } else {
+        span = `<span class="attribution-name">${n}</span>`;
+      }
+
+      if (s.length > 0) {
+        s = `${s} and ${span}`;
+      } else {
+        s = `${span}`;
+      }
+      i += 1;
+    });
+    attributionEL.innerHTML = s;
+  }
+}
+
+function decorateSocialShareList(list) {
```

Comment:
Yes I know there is some code duplication here. I was just going to later move it to a common js file like utils.js. I didn't think a block was necessary since it doesn't look like these are placed just anywhere on the page by authors on the original site. But I can create a block like we have in main branch

Reply:
@icaraps I have moved some of the duplicate code into a separate js file including this.